### PR TITLE
Fix: faster edge builder for tri icosahedron

### PR DESCRIPTION
**Pull request by havardhhaugen**
**Issue:**
The edge building is very slow for graphs with high mesh refinement.

**Fix**
Replaced add_neighbors_edges in add_edges_to_nx_graph with a list-comprehension function that significantly speeds up the mesh building for large meshes. 

I have tested the functionality (in terms of speed) for two setups:
n320 with mesh refinement 7: 2h37min -> 1min8sec 
o96+10km with mesh refinement 5+8: 3min 17sec -> 48sec
n320+2.5km with mesh refinement 7+10: 20min after change (did not check before)

I have checked that the graphs produced for these examples look ok, but I have not done extensive testing to see if this breaks something else (LAM graphs for insance?) 

I also did not test for the case where `vertex_mapping_index=None` and `self_loops=True` as I am 100% sure what the intended behavior should be for these cases.

**Todo**
Clean up the added code, remove print statements and commented code

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-graphs?pullRequest=92) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-graphs?pullRequest=92) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-graphs?pullRequest=92) it.</sub>

**Comment by JPXKQX** [comment]
Good work! I have tested with LAM and it works fine.

File: src/anemoi/graphs/generate/tri_icosahedron.py [hunk 1]
```diff
@@ -270,3 +281,24 @@ def add_neigbours_edges(
             graph.add_edge(node_neighbour, node)
 
     return graph
+
+
+def create_node_neighbors_list(
```
Comment (JPXKQX) [inline src/anemoi/graphs/generate/tri_icosahedron.py:286]: I would probably use neighbours for consistency with the rest of the package and the config file.

**Review by JPXKQX — APPROVED** [review]
Looks good to me! Thanks for the contribution!

**Comment by havardhhaugen** [comment]
Went through this a final time and realized I had a logical error when self loops = True (not sure if self loops currently has a usecase?) Anyways that should be ok now

**Review by JPXKQX — APPROVED** [review]
I don't think anyone uses the `self_loops=True`.
